src/core/buffer.c

```diff
@@ -78,21 +79,21 @@ void janet_buffer_deinit(JanetBuffer *buffer) {
 }
 
 /* Initialize a buffer */
-JanetBuffer *janet_buffer(int32_t capacity) {
+JanetBuffer *janet_buffer(size_t capacity) {
     JanetBuffer *buffer = janet_gcalloc(JANET_MEMORY_BUFFER, sizeof(JanetBuffer));
     return janet_buffer_init_impl(buffer, capacity);
 }
 
 /* Ensure that the buffer has enough internal capacity */
-void janet_buffer_ensure(JanetBuffer *buffer, int32_t capacity, int32_t growth) {
+void janet_buffer_ensure(JanetBuffer *buffer, size_t capacity, size_t growth) {
     uint8_t *new_data;
     uint8_t *old = buffer->data;
     if (capacity <= buffer->capacity) return;
     janet_buffer_can_realloc(buffer);
-    int64_t big_capacity = ((int64_t) capacity) * growth;
-    capacity = big_capacity > INT32_MAX ? INT32_MAX : (int32_t) big_capacity;
+    size_t big_capacity = (capacity) * growth;
```

Comment:
Same issue as before with 32 bit platforms

Reply:
This isn't resolved - for example on 32 bit platforms:

If capacity = 0xBFFFFFFF and growth=2, then big_capacity will underflow after the multiply. We need big_capacity to _always_ be 64 bit, even on 32 bit platforms.



64 bit platforms are safe since we don't use all 64 bits anyway - we stop after about 53 bits due to the requirement that we only use sizes representable as doubles, but that is practically large enough anyway.